### PR TITLE
Server side rotation of cluster bearer tokens

### LINKED ISSUE #1084: Ability to rotate the bearer token used to manage external clusters
We should provide a CLI option to rotate the bearer token that Argo CD uses to manage external clusters. Today, this is a manual process of:

```
kubectl delete secret argocd-manager-token-XXXXXX -n kube-system
argocd cluster add CONTEXTNAME
```

This could be made a convenience command in the CLI.

### PR DESCRIPTION
**Pull request by jessesuen**
Resolves https://github.com/argoproj/argo-cd/issues/1084

This introduces the following command to rotate credentials:
```shell
argocd cluster rotate-auth https://192.168.64.66:8443
```

It is able to perform token rotation server side, using the following logic:
1. get current bearer token to cluster
2. identify secret & namespace associated with token (this information can be extracted from token's JWT claims information)
3. transfer metadata (e.g. service account object reference) to a new empty secret
4. wait for k8s to populate new empty secret with a new token
5. update argocd db with new token
6. delete old secret

Unit tests are pending

### CODE REVIEW

File: cmd/argocd/commands/cluster.go [hunk 1]
```diff
@@ -202,7 +204,7 @@ func NewClusterGetCommand(clientOpts *argocdclient.ClientOptions) *cobra.Command
 // NewClusterRemoveCommand returns a new instance of an `argocd cluster list` command
 func NewClusterRemoveCommand(clientOpts *argocdclient.ClientOptions) *cobra.Command {
 	var command = &cobra.Command{
-		Use:   "rm",
+		Use:   "rm CLUSTER",
```
Comment (alexec) [inline cmd/argocd/commands/cluster.go]: minor - might be good to have an example

File: cmd/argocd/commands/cluster.go [hunk 2]
```diff
@@ -217,7 +219,7 @@ func NewClusterRemoveCommand(clientOpts *argocdclient.ClientOptions) *cobra.Comm
 
 			for _, clusterName := range args {
 				// TODO(jessesuen): find the right context and remove manager RBAC artifacts
```
Comment (alexec) [inline cmd/argocd/commands/cluster.go]: TODO still relevant?

**Review by alexec — COMMENTED** [review]
Can you walk me through this one please? Whenever you are free.

File: server/cluster/cluster.go [hunk 3]
```diff
@@ -174,6 +175,49 @@ func (s *Server) Delete(ctx context.Context, q *cluster.ClusterQuery) (*cluster.
...
+	}
+	newSecret, err := clusterauth.GenerateNewClusterManagerSecret(kubeclientset, claims)
+	if err != nil {
+		return nil, err
+	}
+	// we are using token auth, make sure we don't store client-cert information
+	clust.Config.KeyData = nil
+	clust.Config.CertData = nil
+	clust.Config.BearerToken = string(newSecret.Data["token"])
+	clust, err = s.db.UpdateCluster(ctx, clust)
```
Comment (alexec) [inline server/cluster/cluster.go]: Lint error:
```
server/cluster/cluster.go:209:2: SA4006: this value of `clust` is never used (staticcheck)
	clust, err = s.db.UpdateCluster(ctx, clust)
```

Comment (jessesuen) [inline server/cluster/cluster.go]: Should validate new token works.

File: cmd/argocd/commands/cluster.go (hunk 2 above)
Comment (jessesuen) [reply cmd/argocd/commands/cluster.go]: Yes, we currently leave behind the argocd-manager service account, role, rolebinding during a `cluster rm`. We should eventually have an option during `cluster rm` to delete these artifacts.

File: cmd/argocd/commands/cluster.go (hunk 1 above)
Comment (jessesuen) [reply cmd/argocd/commands/cluster.go]: I'm just fixing the basic usage. Cobra does support a separate section for showing examples, but I don't want to do it for just one sub command.

**Review by alexmt — APPROVED** [review]

**Pull request merged**